### ISSUE TITLE
Latest ID-VG data of nfet_01v8 has unexpected discontinuities in its derivatives

### ISSUE THREAD

**Issue by bmurmann** [google/skywater-pdk-sky130-raw-data#18]
Example:

![image](https://user-images.githubusercontent.com/62622760/185949222-35255227-50e6-4399-b62a-5aa4d01996db.png)

https://colab.research.google.com/drive/1PggoBDvjzV8aN_CluYQCiRV9qptRjlDq#scrollTo=QuGRGGmTCJMi

**Comment by bmurmann** [comment]
The resulting gm and gm/ID look like this (should be smooth):

![image](https://user-images.githubusercontent.com/62622760/185949668-206c6366-341c-4476-9a4f-62fdd84b81c6.png)

**Comment by danchitnis** [comment]
@bmurmann to my experience this occurs when the equipment changes gain to cope with the dynamic range (in this case 1mA). In general, it is very hard to obtain high-quality large dynamic range data, and the measured data needs to be checked at the point of collection, and the equipment to be self-calibrated continuously.

**Comment by bmurmann** [comment]
@danchitnis I fully agree with the need to look at the data during measurement to adress calibration etc. But we know it can be done, as seen e.g. here: https://www.mos-ak.org/munich_2007/papers/02_MOS-AK_Assenmacher.pdf.

**Comment by akinakturk** [comment]
This device was showing this discontinuity. I am not sure if it related to the device itself or the contact (I do not believe that this is an instrument issue). We can measure this again using a different setup and a different die.

**Comment by akinakturk** [comment]
I am planning to measure similar devices on a different die. We need to compile a list of devices that would be good to measure using a different setup and a die.

**Comment by akinakturk** [comment]
If you think you need higher resolution for some select devices, please let me know.

**Comment by danchitnis** [comment]
> This device was showing this discontinuity. I am not sure if it related to the device itself or the contact (I do not believe that this is an instrument issue). We can measure this again using a different setup and a different die.

@akinakturk, if you find a discontinuity during the IV measurements, it would be a good idea to fix the gain of the instruments (this could be *range* based on your instrument) and then sweep around that discontinuity. This will give you more indication if it is the device or the instrument. 

> I am planning to measure similar devices on a different die. We need to compile a list of devices that would be good to measure using a different setup and a die.

How many repeats do you have per device? when measuring a device, please also record its location inside the chip, as this will help for matching analysis etc.

**Comment by akinakturk** [comment]
The location of the device inside the chip is tracked using the module number and the pad numbers (written at the end of the device name in parenthesis) . There is only one of each module on this particular chip.

Range is typically set to auto. This is almost universally true for all curve tracers / parametric analyzers. I can limit the range but it is impractical in most applications. I use either HP4155 or B1500. I find HP4155 better in terms of noise and accuracy even though it is a much older version of B1500. I literally tested thousands and thousands of devices using similar settings in the past. The current is relatively large in this case. This makes it less likely to be due to a range issue. In any case, using a different die might resolve the issue. By the way, if the measurement "appeared" to have issues, we retook the data. These are not single shot measurements.

**Comment by akinakturk** [comment]
I remeasured this device using a completely different setup and a different die. Even though the peak current of this device on a different die is slightly higher, it still has the same discontinuity. this is coming from the device itself. Please see the attached file (the extension is changed from mdm to txt to be able to attach it here)


[5290_12_IDVG_Finer.txt](https://github.com/google/skywater-pdk-sky130-raw-data/files/9552831/5290_12_IDVG_Finer.txt)

**Comment by QuantamHD** [comment]
If the device itself has an unexpected curve we should leave this issue open to triage. @bmurmann thoughts?

**Comment by akinakturk** [comment]
By the way, in the document describing the connections, only three terminals are marked for the devices in this module: b, g and d.  The measurements seem to indicate b and s are shorted (otherwise we cannot measure any IV curves).

**Comment by bmurmann** [comment]
@akinakturk The discontinuity is actually worse in this new measurement; now the current itself is also discontinuous, not just its derivative. This behavior is physically not possible (if was, please explain, and let's write award-winnig paper... 8)

**Comment by bmurmann** [comment]
Have you taken in forward and backward direction of the VG sweep to see if they match?

**Comment by hung-chi** [comment]
Hi,

This is Hung-Chi from EPFL. We just received some test structures.
We are planning to do the measurement on the devices that were reported due to the poor data quality.

However, I have a problem of finding a device from the chip. When I look at the [spread sheet](https://docs.google.com/spreadsheets/d/1LSINFWJRTvHEC7PzT-Y1yHR9Nuz75gVvBY0e7oAhnCg/edit#gid=25336723). I only get the number of row die. I couldn't know which column has the device I look for.

Thanks in advance for the answer.

Hung-Chi

**Comment by QuantamHD** [comment]
@mithro ^^^

**Comment by QuantamHD** [comment]
@hung-chi See this diagram on the spreadsheet. Each row of the spreadsheet represents one row of 12 pins left to right.

<img width="963" alt="image" src="https://user-images.githubusercontent.com/3784748/195715111-78d37296-e890-4ed6-8adc-4038814a4b01.png">
